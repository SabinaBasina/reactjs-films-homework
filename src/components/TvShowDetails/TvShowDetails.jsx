import React, { Component } from 'react';
import http from 'axios';
import styles from './TvShowDetails.scss';
import Loading from '../TvShowPage/Loading.gif';
import NoImage from '../TvShow/NoImage.jpg';
import Episodes from '../Episodes/Episodes';

class TvShowDetails extends Component {
  constructor(props) {
    super(props);

    this.state = {
      tvShow: {},
      isReady: false,
    };

    // eslint-disable-next-line react/prop-types
    const { id } = this.props;

    http
      .get(`https://api.tvmaze.com/shows/${id}`)
      .then((response) => {
        this.setState({ tvShow: response.data, isReady: true });
      });
  }


  render() {
    const { isReady, tvShow } = this.state;
    return (
      <div>

        {!isReady
           && (
           <img
             src={Loading}
             className={styles.Ready}
             alt=""
           />
           )}

        {isReady

            && (
            <div className={styles.TvShow}>

              <h1>{tvShow.name}</h1>

              <div className={styles.Information}>

                <img
                  src={tvShow.image === undefined ? NoImage : tvShow.image.medium}
                  className={styles.TvShowImage}
                  alt=""
                />

                <div>
                  <p dangerouslySetInnerHTML={{ __html: tvShow.summary }} />
                  <br />
                  <b>Genres:</b>
                  {' '}
                  {tvShow.genres && tvShow.genres.join(', ')}
                  <br />
                  <b>Country:</b>
                  {' '}
                  {tvShow.network && tvShow.network.country.name}
                  <br />
                  <b>Premiered:</b>
                  {' '}
                  <time>{tvShow.premiered}</time>
                </div>

              </div>


              <p className={styles.Episodes}>Episodes</p>
              <Episodes nameTvShow={tvShow.name} />

            </div>
            )
        }

      </div>
    );
  }
}

export default TvShowDetails;
