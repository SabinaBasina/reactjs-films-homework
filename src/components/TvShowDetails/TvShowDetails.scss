.Ready {
  display: block;
  margin: 30px auto 0 auto;
  text-align: center;
}

.TvShow {
  text-align: justify;
}

.TvShow h1 {
  text-align: center;
  margin-top: 40px;
  margin-bottom: 20px;
  font-weight: 500;
}

.Information {
  display: block;
  margin-left: 20%;
  margin-right: 20%;
}

.Information::after {
  content: " ";
  display: table;
  clear: both;
}

.TvShowImage {
  float: left;
  margin: 0 35px 15px 0;
}

.Episodes {
  margin: 100px 15% 60px 15%;
  height: 35px;
  background-color: #222;
  border-radius: 8px;
  color: white;
  font-weight: 100;
  font-size: 24px;
  text-align: center;
}

@media only screen and (max-width: 1100px) {
  .Information {
    margin-left: 10%;
    margin-right: 10%;
  }

  .Episodes {
    margin-left: 5%;
    margin-right: 5%;
  }
}

@media only screen and (max-width: 820px) {
  .Episodes {
    margin-left: 0;
    margin-right: 0;
  }

  .TvShowImage {
    display: block;
    margin: 0 auto 40px auto;
    float: none;
  }

  .Information {
    margin-left: 5%;
    margin-right: 5%;
  }
}
