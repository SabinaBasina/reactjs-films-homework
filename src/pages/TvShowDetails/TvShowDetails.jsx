/* eslint-disable react/no-danger */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import styles from './TvShowDetails.scss';
import NoImage from '../../images/NoImage.jpg';
import Episodes from '../../components/Episodes';
import Loader from '../../components/Loader';

class TvShowDetails extends Component {
  componentDidMount() {
    const { id, loadTvShowsDetails } = this.props;
    loadTvShowsDetails(id);
  }

  render() {
    const { tvShow, isReadyTvShow } = this.props;
    return (
      <div>

        <Loader loading={!isReadyTvShow} />

        {isReadyTvShow
            && (
            <div className={styles.tvshow}>

              <h1>{tvShow.name}</h1>

              <div className={styles.information}>

                <img
                  src={tvShow.image ? tvShow.image.medium : NoImage}
                  className={styles.image}
                  alt=""
                />

                <div>
                  <p dangerouslySetInnerHTML={{ __html: tvShow.summary }} />
                  <br />
                  <b>Genres: </b>
                  {tvShow.genres && Array.prototype.join.call(tvShow.genres, ', ')}
                  <br />
                  <b>Country: </b>
                  {tvShow.network && tvShow.network.country.name}
                  <br />
                  <b>Premiered: </b>
                  <time>{tvShow.premiered}</time>
                </div>

              </div>


              <p className={styles.episodes}>Episodes</p>
              <Episodes nameTvShow={tvShow.name} />

            </div>
            )
        }

      </div>
    );
  }
}

export default TvShowDetails;

TvShowDetails.propTypes = {
  id: PropTypes.string,
  loadTvShowsDetails: PropTypes.func,
  isReadyTvShow: PropTypes.bool,
  tvShow: PropTypes.instanceOf(Object),
};

TvShowDetails.defaultProps = {
  loadTvShowsDetails: () => { },
  id: 'undefined',
  isReadyTvShow: false,
  tvShow: undefined,
};
