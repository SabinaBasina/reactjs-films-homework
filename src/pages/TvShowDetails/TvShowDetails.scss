.tvshow {
  text-align: justify;
}

.tvshow h1 {
  text-align: center;
  margin-top: 30px;
  margin-bottom: 20px;
  font-weight: 500;
}

.information {
  display: block;
  margin-left: 20%;
  margin-right: 20%;
}

.information::after {
  content: " ";
  display: table;
  clear: both;
}

.image {
  float: left;
  margin: 0 35px 15px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.addTvShow,
.loadingTvShow {
  outline: none;
  width: 130px;
  height: 40px;
  margin-top: 15px;
  border-radius: 50px;
  background-color: white;
  font-size: 16px;
  font-weight: 400;
}

.addTvShow {
  cursor: pointer;
}

.addTvShow:hover {
  background-color: rgba(150, 187, 187, 0.178);
}

.episodes {
  margin: 80px 15% 60px 15%;
  height: 35px;
  background-color: #222;
  border-radius: 8px;
  color: white;
  font-weight: 100;
  font-size: 24px;
  text-align: center;
}

@media only screen and (max-width: 1100px) {
  .information {
    margin-left: 10%;
    margin-right: 10%;
  }

  .episodes {
    margin-left: 5%;
    margin-right: 5%;
  }
}

@media only screen and (max-width: 820px) {
  .episodes {
    margin-left: 0;
    margin-right: 0;
  }

  .image {
    margin: 0 auto 40px auto;
    float: none;
  }

  .information {
    margin-left: 5%;
    margin-right: 5%;
  }
}