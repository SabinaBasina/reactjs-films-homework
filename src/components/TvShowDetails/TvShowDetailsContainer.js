import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import TvShowDetails from './TvShowDetails';
import { loadTvShowsDetails } from '../../redux/actions/tvshowdetails';


const mapStateToProps = createStructuredSelector({
  tvShow: state => state.tvshowdetails.tvShow,
});

const mapDispatchToProps = {
  loadTvShowsDetails,
};

export default connect(mapStateToProps, mapDispatchToProps)(TvShowDetails);
