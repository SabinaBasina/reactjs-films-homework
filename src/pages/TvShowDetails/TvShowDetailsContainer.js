import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import TvShowDetails from './TvShowDetails';
import { loadTvShowsDetails } from '../../modules/moduleDetails/moduleDetailsActions';
import { getTvShow, getIsReadyTvShow } from '../../modules/moduleDetails/moduleDetailsSelectors';

const mapStateToProps = createStructuredSelector({
  tvShow: getTvShow,
  isReadyTvShow: getIsReadyTvShow,
});

const mapDispatchToProps = {
  loadTvShowsDetails,
};

export default connect(mapStateToProps, mapDispatchToProps)(TvShowDetails);
