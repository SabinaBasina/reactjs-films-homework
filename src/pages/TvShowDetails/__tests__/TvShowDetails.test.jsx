import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Renderer from 'react-test-renderer';

import TvShowDetails from '../TvShowDetails';

describe('test', () => {
  it('TvShowDetails test 1', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TvShowDetails />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it('TvShowDetails test 2', () => {
    const renderer = Renderer.create(<TvShowDetails id="1" />);
    const testInstance = renderer.toJSON();
    expect(testInstance).toMatchSnapshot();
  });

  test('Episode test img is', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TvShowDetails tvShow={{ image: { medium: 'medium' } }} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  test('Episode test img no', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TvShowDetails tvShow={{ }} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  test('Episode test genres is', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TvShowDetails tvShow={{ genres: 'genres' }} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  test('Episode test country is', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TvShowDetails tvShow={{ network: { country: { name: 'name' } } }} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
