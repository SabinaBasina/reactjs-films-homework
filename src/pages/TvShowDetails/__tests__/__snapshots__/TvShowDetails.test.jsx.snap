// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShowDetails Episode test country name is 1`] = `
<div>
  <_default
    loading={false}
  />
  <div
    className="tvshow"
  >
    <h1 />
    <div
      className="information"
    >
      <img
        alt=""
        className="image"
        src="test-file-stub"
      />
      <div>
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
        <br />
        <b>
          Genres: 
        </b>
        <br />
        <b>
          Country: 
        </b>
        name
        <br />
        <b>
          Premiered: 
        </b>
        <time />
      </div>
    </div>
    <p
      className="episodes"
    >
      Episodes
    </p>
    <Connect(Episodes) />
  </div>
</div>
`;

exports[`TvShowDetails Episode test genres is 1`] = `
<div>
  <_default
    loading={false}
  />
  <div
    className="tvshow"
  >
    <h1 />
    <div
      className="information"
    >
      <img
        alt=""
        className="image"
        src="test-file-stub"
      />
      <div>
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
        <br />
        <b>
          Genres: 
        </b>
        g, e, n, r, e, s
        <br />
        <b>
          Country: 
        </b>
        <br />
        <b>
          Premiered: 
        </b>
        <time />
      </div>
    </div>
    <p
      className="episodes"
    >
      Episodes
    </p>
    <Connect(Episodes) />
  </div>
</div>
`;

exports[`TvShowDetails Episode test img is 1`] = `
<div>
  <_default
    loading={false}
  />
  <div
    className="tvshow"
  >
    <h1 />
    <div
      className="information"
    >
      <img
        alt=""
        className="image"
        src="medium"
      />
      <div>
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
        <br />
        <b>
          Genres: 
        </b>
        <br />
        <b>
          Country: 
        </b>
        <br />
        <b>
          Premiered: 
        </b>
        <time />
      </div>
    </div>
    <p
      className="episodes"
    >
      Episodes
    </p>
    <Connect(Episodes) />
  </div>
</div>
`;

exports[`TvShowDetails Episode test tvShow no 1`] = `
<div>
  <_default
    loading={false}
  />
  <div
    className="tvshow"
  >
    <h1 />
    <div
      className="information"
    >
      <img
        alt=""
        className="image"
        src="test-file-stub"
      />
      <div>
        <p
          dangerouslySetInnerHTML={
            Object {
              "__html": undefined,
            }
          }
        />
        <br />
        <b>
          Genres: 
        </b>
        <br />
        <b>
          Country: 
        </b>
        <br />
        <b>
          Premiered: 
        </b>
        <time />
      </div>
    </div>
    <p
      className="episodes"
    >
      Episodes
    </p>
    <Connect(Episodes) />
  </div>
</div>
`;

exports[`TvShowDetails TvShowDetails snapshot 1`] = `
<div>
  <_default
    loading={true}
  />
</div>
`;

exports[`TvShowDetails TvShowDetails test id is 1`] = `
<div>
  <img
    alt="Loading"
    className="loader"
    src="test-file-stub"
  />
</div>
`;
